src/VscPreviewExtension/src/tokenTreeContentProvider.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { workspace, ExtensionContext, TextDocumentContentProvider, EventEmitter, Event, Uri } from "vscode";
+import * as path from "path";
+
+export class TokenTreeContentProvider implements TextDocumentContentProvider {
+    // TODO: extra the common pattern form this with MarkdownDocumentContentProvider
+    private _context: ExtensionContext;
+    private _onDidChange = new EventEmitter<Uri>();
+    private _jsonContent = "";
+    private _port;
+
+    constructor(context: ExtensionContext, port) {
+        this._port = port;
+        this._context = context;
+    }
+
+    private getMediaPath(mediaFile): string {
+        return this._context.asAbsolutePath(path.join("media", mediaFile));
+    }
+
+    public provideTextDocumentContent(uri: Uri): Thenable<string> {
+        return workspace.openTextDocument(Uri.parse(uri.query)).then(document => {
+            const content = [
+                "<!DOCTYPE html>",
+                "<html>",
+                "<head>",
+                `<meta http-equiv="Content-type" content="text/html;charset=UTF-8">`,
+                `<link rel="stylesheet" type="text/css" href="${this.getMediaPath(`token.css`)}" >`,
+                `</head>`,
+                `<body><!-- ` + this._jsonContent.substring(0, this._jsonContent.length - 1) + `--><!--` + this._port.toString() + `-->`,
```

Comment:
to inform the js which port I will use

---
In reply to: [77942161](https://github.com/dotnet/docfx/pull/664#discussion_r77942161) [](ancestors = 77942161)